extensions/parquet/table/src/main/java/io/deephaven/parquet/table/pagestore/topage/ToBigDecimalPage.java

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.parquet.table.pagestore.topage;
+
+import io.deephaven.chunk.ChunkType;
+import io.deephaven.chunk.attributes.Any;
+import io.deephaven.parquet.base.BigDecimalParquetBytesCodec;
+import io.deephaven.util.channel.SeekableChannelContext;
+import io.deephaven.util.codec.ObjectCodec;
+import org.apache.parquet.column.Dictionary;
+import org.jetbrains.annotations.NotNull;
+
+import java.math.BigDecimal;
+import java.util.function.Function;
+
+public class ToBigDecimalPage<ATTR extends Any> implements ToPage<ATTR, BigDecimal[]> {
+
+    private static final ToBigDecimalPage<? extends Any> INSTANCE = new ToBigDecimalPage<>();
+
+    public static <ATTR extends Any> ToPage<ATTR, BigDecimal[]> create(
+            final Class<?> nativeType,
+            final int precision,
+            final int scale,
+            final Function<SeekableChannelContext, Dictionary> dictionarySupplier,
+            final int encodedSizeInBytes) {
+        if (BigDecimal.class.equals(nativeType)) {
+            if (dictionarySupplier == null) {
+                // noinspection unchecked
+                return (ToPage<ATTR, BigDecimal[]>) INSTANCE;
+            }
+            // Codec is only used for dictionary encoding
+            // TODO Currently codec is always made, even if dictionary is NONE
```

Comment:
TODO

Reply:
You could change the `ColumnChunkReaderImpl` to determine earlier if there's a dictionary, but it's not obvious that it's worth the change.